ViewBase.js

```diff
@@ -112,25 +112,25 @@ define(["require", "dojo/when", "dcl/dcl", "dojo/Deferred", "dapp/utils/view"],
 				// tags:
 				//		private
 				//
-				var viewControllerDef = new Deferred();
 				var path;
 
 				//TODO: There is a problem with the order of views being added if no controller is listed
 				//TODO: Seems like order problem can be solved by using ViewBase if it is not set, try this for now!
-				if (!this.controller) {
-					this.controller = "dapp/ViewBase";
-				}
-				if (!this.controller) {
-					viewControllerDef.resolve(true);
-					return viewControllerDef;
-				} else {
-					path = this.controller.replace(/(\.js)$/, "");
-				}
+				return new Promise(function (resolve) {
+					if (!this.controller) {
+						this.controller = "dapp/ViewBase";
+					}
+					if (!this.controller) {
+						resolve(true);
+						return;
+					} else {
+						path = this.controller.replace(/(\.js)$/, "");
+					}
 
-				require([path], function (controller) {
-					viewControllerDef.resolve(controller);
-				});
-				return viewControllerDef;
+					require([path], function (controller) {
+						resolve(controller);
+					});
```

Comment:
Right, well since the first if will set this.controller if it is not set, I can remove the if and the else, and just leave the path = this.controller.replace(/(\.js)$/, "");